docs/research.md

```diff
@@ -1,33 +1,85 @@
+Cons:
+- Does not work out of the box.
+- Hardcoded build process limits flexibility regarding version attachment.
+- Contracts are deployed from the same release used to build the binaries, which can create complications.
 
-- CLI for local self-service
+Source:
+- https://github.com/upnodedev/opstack-compose
+
+### optimism/op-deployer
+
+Pros:
+- Developed and maintained by Optimism itself
+- Utilizes the [OPCM](https://github.com/ethereum-optimism/design-docs/blob/main/protocol/op-contracts-manager-arch.md) which simplifies contract deployments significantly.
+
+Cons:
+- Still in the early stages for our use; limited documentation and usage examples.
+- Primarily focused on contract deployment.
+- Currently relies on a deployment script.
+- Does not address infrastructure deployment and management.
 
-- Deploy with helm on top of `opstack-compose`'s docker-compose
-	- Networking already solved
-	- Cloud provider-agnostic
-	- Provides scalability
-    - Eases deploy process
+Source:
+- https://docs.optimism.io/builders/chain-operators/tools/op-deployer
 
+# Proposed solutions
+
+We have decided to develop our own CLI and infrastructure tooling to create a more flexible deployment process and address some of the inconveniences we encountered while testing existing solutions. Given the challenges in getting current solutions to work easily and the relatively short time required to replicate them, we aim for our development to also meet new requirements, including a user interface and scalable deployment.
+
+## Decisions and drivers
+
+- We chose Rust for its performance, cost efficiency, and growing popularity, ensuring reliable and scalable solutions.
+- The CLI will be an implementation of a core Rust package that manages the business logic for creating the blockchain. This approach allows us to maintain testable code while also enabling multiple client implementations, including the server itself.
+- CLI for local self-service
+- We will deploy using `Helm` based on `opstack-compose`. Leveraging `Kubernetes` enables us to expand the number of providers that can support us while providing the scalability we need. This approach will simplify the user experience during deployment. We may incorporate `Terraform` to further facilitate the process.
 - Cloud service App: UI + Lambda API implementing Core
-	- Rust lambdas are cheap and scale easily
\ No newline at end of file
+- Blockexplorer. For the block explorer [Blockscout](https://www.blockscout.com/) appears to be the best choice. BlockScout is an open-source block explorer that supports OP Stack chains, is recommended by the Optimism documentation, and is widely implemented in many current solutions.
+
+# Risk and uncertainties 
+
+1. It must be sufficiently extensible to handle new features.
+2. Upgrades management 
+3. Contemplate testing environments. 
+4. Avoid overloading user with enviroment setup complexities.
+5. Testnet tokens availability and proper config setup
+
+ 
+
+ 
```

Comment:
Maybe run a formatter to remove unnecessary lines (?)

Reply:
Nice! Done 👍🏽